vault/identity_store_entities.go

```diff
@@ -717,7 +725,16 @@ func (i *IdentityStore) handlePathEntityListCommon(ctx context.Context, req *log
 	return logical.ListResponseWithInfo(keys, entityInfo), nil
 }
 
-func (i *IdentityStore) mergeEntity(ctx context.Context, txn *memdb.Txn, toEntity *identity.Entity, fromEntityIDs []string, force, grabLock, mergePolicies, persist bool) (error, error) {
+// A small type to help error reporting during entity alias clashes
+type aliasClashInformation struct {
+	toEntityId    string
+	fromEntityId  string
+	toAliasId     string
+	fromAliasId   string
+	mountAccessor string
+}
+
+func (i *IdentityStore) mergeEntity(ctx context.Context, txn *memdb.Txn, toEntity *identity.Entity, fromEntityIDs, conflictingAliasIDsToKeep []string, force, grabLock, mergePolicies, persist, forceMergeAliases bool) (error, error) {
```

Comment:
I know there were already a ton of params before you got here, but I can't look at such a long list of params and not ask: can we make a struct for these?  Especially when there a a bunch of bools like this it's pretty hard to read.

Reply:
We could, but it doesn't seem like a huge value add to me, we'd just be creating a new struct and passing that in the two places this function is called

In other words, instead of:
```
merge(arg1 type1, arg2 type2, arg3 type3, ...)
//and to call
merge(arg1, arg2, arg3)
```
we'd have something like
```
merge(mergeArgs  *mergeArgs)
//and to call
args := &mergeArgs{
arg1: arg1,
arg2: arg2,
arg3: arg3,
...
}
merge(args)
```

In other words, I guess I don't feel like a struct would stop the function from being unwieldy - it's unwieldy either way and the 'number of arguments' doesn't really change. It would have been great if we could have joined the 'force' functionality and reduced the number of arguments by one, but alas.

Reply:
When I see a call like `func(a, b, c, false, true, false)`, and then have to consult the function definition to figure out what those are, it slows me down.  That's why I think it's more readable to use a struct, assuming it's initialized using field names.